### DISCUSSION TITLE
Switch running session between wayvnc headless and physical seat

### DISCUSSION THREAD

**Discussion by headbank** [any1/wayvnc#332] [General]
I'll start by recognising that I might be the only person on Earth interested in something like this. Second, posting in "General" because I suspect its scope exceeds WayVNC alone (although it's a key part).

I have been using WayVNC + WayFire as the only available way to use my laptop at my office workstation. I use headless because the workstation monitor has a different screen-size and I want the session to fit it. (Also I don't want the physical display to be active when the laptop is not near me.)

To set this up I have to logout my physical session, go to a VT and run the headless session and start WayVNC. It'd be a lot cooler if I could start a headless output (and virtual inputs) within the existing session, create a wayvnc instance for them, move the panel and any running client app windows onto there, and switch off the physical screen (that last part could be optional I guess). Then later the process could be reversed to return to a regular physical session.

I'm just interested in learning whether something like this is architecturally possible with wayvnc and wlroots as they stand; whether they or compositors would have to implement further features to permit it. It seems like something that might be within reach or even already possible, but I don't know enough to fully get a sense of it. Any clues or insight would be appreciated.

**Comment by any1** [comment]
If you patch wayfire like so https://github.com/raspberrypi-ui/wayfire/pull/6, you can turn off your screen and then your session will fall back to headless.

**Reply by headbank** [reply]
Thanks, that's very encouraging to hear! The linked discussion challenges my level of background knowledge quite a bit, so I have a couple of followups if I may:

1. I see that the merge is to RPi's fork of wayfire; Would I expect the commit to apply cleanly to stock wayfire-0.9.0 sources (current Gentoo release)? Is the content of [this commit](https://github.com/raspberrypi-ui/wayfire/commit/91ffd31248e529b8878ca7260250b7ae1b9fe786) really all I need?
2. Can you explain about the `auto-headless` plugin that is mentioned there? I cannot find any other references to this anywhere and I'm unclear which involved component it belongs to (wayfire / wlroots / wayvnc) or even if that part of the discussion remains valid.
3. Where/when does wayvnc fit into this: would you have it running before the switch, or must it be started manually afterwards?
4. (This might be outside of your wheelhouse but) how do I turn off a laptop screen manually?
